### DISCUSSION TITLE
Scene Viewer and iOS?

### DISCUSSION THREAD

**Discussion by jakecjones** [google/model-viewer#2042] [Ideas]
### Description

I just noticed that I can use Scene Viewer in the browser on iOS. After walking through the Augmented Reality demo on the model-viewer examples, it doesn't give me an option to launch scene viewer if I am on an iOS device. Not sure how to test specifically with model-viewer but it would be worth it to know if scene viewer can now be launched from an iOS device with model-viewer.

#### Live Demo
N/A

### Version
N/A

### Browser Affected

- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by jakecjones** [comment]
HOW TO TEST (outside of model-viewer):  google search for "wolf" on safari or google chrome on an iOS device. You should be able to see a "view in 3d" option. 

![IMG_BC0EECCE39DA-1](https://user-images.githubusercontent.com/31197056/108730943-0a657780-74fa-11eb-9fb6-f047b0e407fa.jpeg)

When you click on this it will open up in scene viewer, not iOS ar quicklook.

**Comment by elalish** [comment]
I believe this only works if you have the iOS Google app installed (of which SceneViewer is now a part). Still, I agree that we should probably enable launching it from model-viewer.

**Reply by jakecjones** [reply]
Good clarification, I didn't notice that at first. For model-viewer, It sounds like this would also be a good solution for a model that only has a glb format associated. That way you would still get the option to use AR.

**Reply by elalish** [reply]
Unfortunately, it looks like opening sceneViewer on iOS with arbitrary GLBs is blocked by security.

**Comment by milesgreen** [comment]
This is just from my observations:

Previously, "_View in 3D_" from a Google Search result in Safari iOS would open the native iOS AR Quick Look viewer.

Now, "_View in 3D_" opens the iOS Google app (if you have it) which contains Scene Viewer.

If you don't have the Google app on iOS, then "_View in 3D_" from Search results links to a URL, still within Safari. You see the 3D model, but the "_View in your space_" button is replaced with "_View in your space with Google app_" which links you to the Google app on the Apple App Store.

Seems a shame that iOS users will potentially get pushed to install an app from the Search results, when an AR viewer is already built into iOS. Maybe this is to ensure parity between experiences (USDZ has different features to GLTF), or maybe it's to reduce burden on content owners so they don't have to produce 2 assets? Schema.org already defined a 3DModel type to handle multiple encodings though, which I had assumed Search would use for indexing. I guess Apple should just get on board with support for GLTF and WebXR already. 

Totally understanding that this isn't the Model Viewer team's remit though 👍

-----

**_From Safari iOS when Google app is installed:_**

![IMG_8981](https://user-images.githubusercontent.com/53870482/109007028-96ff6980-76a3-11eb-8490-7fa0ef1f4ca9.PNG)

-----

**_From Safari iOS when Google app is not installed:_**

![IMG_8982](https://user-images.githubusercontent.com/53870482/109007081-a8487600-76a3-11eb-8d7a-0b9357382315.PNG)

**Reply by elalish** [reply]
Agreed that's a weird experience; any thoughts @tspiaki? Also agree that the real way forward here is WebXR in Safari. Hopefully before the heat death of the universe.
